### PR TITLE
GG-27188 JMX metric indexbuildcountpartitionsleft show values below zero

### PR DESCRIPTION
**Pull request by a-polyakov**

### CODE REVIEW

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/CacheGroupMetricsImpl.java [hunk 1]
```diff
@@ -191,6 +191,13 @@ public void setIndexBuildCountPartitionsLeft(long idxBuildCntPartitionsLeft) {
     public void decrementIndexBuildCountPartitionsLeft() {
         idxBuildCntPartitionsLeft.decrement();
     }
+    /**
```
Comment (zstan) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/cache/CacheGroupMetricsImpl.java]: new line need to be here.

File: modules/indexing/src/test/java/org/apache/ignite/internal/processors/cache/CacheGroupMetricsWithIndexTest.java [hunk 2]
```diff
@@ -79,8 +85,32 @@
...
+                ArrayList<QueryIndex> indexes = new ArrayList<>();
+
+                indexes.add(new QueryIndex(COLUMN1_NAME));
+
+                indexes.add(new QueryIndex(COLUMN2_NAME));
+
+                qryEntity.setIndexes(indexes);
+
+                cacheCfg.setQueryEntities(Collections.singletonList(qryEntity));
+            }else if (GROUP_NAME.equals(cacheCfg.getGroupName()) && CACHE_NAME3.equals(cacheCfg.getName())) {
```
Comment (zstan) [inline modules/indexing/src/test/java/org/apache/ignite/internal/processors/cache/CacheGroupMetricsWithIndexTest.java]: and new line here

File: modules/indexing/src/test/java/org/apache/ignite/internal/processors/cache/CacheGroupMetricsWithIndexTest.java [hunk 3]
```diff
@@ -182,34 +213,34 @@ public void testIndexCreateCountPartitionsLeft() throws Exception {
...
 
         MetricRegistry metrics = cacheGroupMetrics(0, GROUP_NAME).get2();
 
         GridTestUtils.runAsync(() -> {
             String createIdxSql = "CREATE INDEX " + INDEX_NAME + " ON " + TABLE + "(" + COLUMN3_NAME + ")";
 
-            cache1.query(new SqlFieldsQuery(createIdxSql)).getAll();
+            cache2.query(new SqlFieldsQuery(createIdxSql)).getAll();
 
             String selectIdxSql = "select * from information_schema.indexes where index_name='" + INDEX_NAME + "'";
```
Comment (ygerzhedovich) [inline modules/indexing/src/test/java/org/apache/ignite/internal/processors/cache/CacheGroupMetricsWithIndexTest.java:241]: Will be better to rewrite it to use our system view. `INFORMATION_SCHEMA` don't support by IGNITE at all.

File: modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/IgniteH2Indexing.java [hunk 4]
```diff
@@ -2024,6 +2024,9 @@ public UpdateResult executeUpdateOnDataNode(
 
         GridFutureAdapter<Void> rebuildCacheIdxFut = new GridFutureAdapter<>();
 
+        if (cctx.group().metrics() != null)
```
Comment (ygerzhedovich) [inline modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/IgniteH2Indexing.java:2027]: when metrics could be null?

**Pull request merged**